types/relay-compiler/language/RelayLanguagePluginInterface.d.ts

```diff
@@ -22,12 +22,14 @@ export type FormatModule = (
       | typeof RelayConcreteNode.FRAGMENT
       | typeof RelayConcreteNode.REQUEST
       | typeof RelayConcreteNode.BATCH_REQUEST
+      | "ConcreteFragment"
+      | "ConcreteRequest"
+      | "ReaderFragment"
       | null;
     docText: string | null;
     concreteText: string;
     typeText: string;
-    hash: string | null;
-    devOnlyAssignments: string | null;
+    devOnlyAssignments?: string | null;
```

Comment:
i saw a PR about this, so i left it in, but made it optional so ts won't yell at me

Reply:
According to the upstream interface, if my rusty Flow isn’t failing me, this property isn’t optional but does take `null` or `undefined` https://github.com/facebook/relay/blob/c0376a5bb06e0e1c3b62d662027b18c0322e84ec/packages/relay-compiler/language/RelayLanguagePluginInterface.js#L176

I’d like our TS version of the interface to accurately mimic the upstream version.

Reply:
ah, i didn't know there was a ref doc. should i put hash back in & add that to the tests? looks like there's some ambiguity with hash & sourceHash in the doc

Reply:
Saw you already did so, thanks!